tests/integration/src/fixtures/module-resolution/index.ts

```diff
@@ -0,0 +1,25 @@
+const allowedPaths = new Set([
+	'/require-ext',
+	'/require-no-ext',
+	'/require-json',
+	'/cloudflare-imports',
+
+	'/third-party/react',
+	'/third-party/remix',
+	'/third-party/discord-api-types',
+	'/third-party/slash-create',
+]);
+
+export default {
+	async fetch(request: Request, env: any) {
+		const url = new URL(request.url);
+		const path = url.pathname;
+
+		if (allowedPaths.has(path)) {
+			const mod = await import(/* @vite-ignore */ `./src${path}`);
+			return Response.json(mod.default);
+		}
+
+		return new Response(`path not found: '${path}'`, { status: 404 });
+	},
+};
```

Comment:
Let's use the `ExportedHandler` type here:

```suggestion
export default {
	async fetch(request) {
		const url = new URL(request.url);
		const path = url.pathname;

		if (allowedPaths.has(path)) {
			const mod = await import(/* @vite-ignore */ `./src${path}`);
			return Response.json(mod.default);
		}

		return new Response(`path not found: '${path}'`, { status: 404 });
	},
} satisfies ExportedHandler;
```